### PR TITLE
Device id option for TF backend of DNN filter

### PR DESCRIPTION
**Pull request by cyberj0g**
See [Issue 1997](https://github.com/livepeer/go-livepeer/issues/1997)

### CODE REVIEW

File: libavfilter/vf_livepeer_dnn.c [hunk 1]
```diff
@@ -82,7 +82,7 @@ static av_cold int init(AVFilterContext *context)
         ctx->logfile = NULL;
         av_log(ctx, AV_LOG_INFO, "output file for log is not specified\n");
     }
-
+    printf("backend_config: %s\n", ctx->dnnctx.backend_options, 30);
```
Comment (jailuthra) [inline libavfilter/vf_livepeer_dnn.c]: Let's clean up this log if it's not needed.

File: libavfilter/dnn_filter_common.h [hunk 2]
```diff
@@ -44,8 +44,7 @@ typedef struct DnnContext {
     { "output",             "output name of the model",   OFFSET(model_outputname), AV_OPT_TYPE_STRING,    { .str = NULL }, 0, 0, FLAGS },\
     { "backend_configs",    "backend configs",            OFFSET(backend_options),  AV_OPT_TYPE_STRING,    { .str = NULL }, 0, 0, FLAGS },\
     { "options",            "backend configs",            OFFSET(backend_options),  AV_OPT_TYPE_STRING,    { .str = NULL }, 0, 0, FLAGS },\
-    { "async",              "use DNN async inference",    OFFSET(async),            AV_OPT_TYPE_BOOL,      { .i64 = 1},     0, 1, FLAGS},
-
+    { "async",              "use DNN async inference",    OFFSET(async),            AV_OPT_TYPE_BOOL,      { .i64 = 1},     0, 1, FLAGS}
```
Comment (jailuthra) [inline libavfilter/dnn_filter_common.h]: Let's avoid formatting changes to common modules, will make it harder to rebase with upstream later on.

File: libavfilter/vf_dnn_processing.c [hunk 3]
```diff
@@ -53,7 +53,7 @@ static const AVOption dnn_processing_options[] = {
 #if (CONFIG_LIBOPENVINO == 1)
     { "openvino",    "openvino backend flag",      0,                        AV_OPT_TYPE_CONST,     { .i64 = 2 },    0, 0, FLAGS, "backend" },
 #endif
-    DNN_COMMON_OPTIONS
+    DNN_COMMON_OPTIONS,
```
Comment (jailuthra) [inline libavfilter/vf_dnn_processing.c]: Let's avoid formatting changes to common modules, will make it harder to rebase with upstream later on.

**Review by jailuthra — APPROVED** [review]
LGTM, except some formatting/cleanup changes.

We should squash & merge this, changing the final commit name to something that follows the repository convention - like `avfilter/dnn: add device_id option in TF backend`

Once this PR is merged, let's update the `install_ffmpeg.sh` scripts in https://github.com/livepeer/lpms/pull/264 and https://github.com/livepeer/go-livepeer/pull/2038 with the new commit ID.

File: libavfilter/vf_dnn_processing.c (hunk 3 above)
Comment (cyberj0g) [reply libavfilter/vf_dnn_processing.c]: Fixed

**Pull request merged**